### PR TITLE
Bug 2091110: Skip ErrImagePull for expected failures

### PR DESCRIPTION
**Pull request by DennisPeriquet**
There are certain [documented exceptions](https://github.com/openshift/origin/blob/f88be40a20026986abd239bfb161f21dce283986/test/extended/util/image/README.md#kube-exceptions) where the ErrImagePull event is expected as part of a negative test.  This PR skips those so they are not counted as failures.

See the [Example CI job](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27202/pull-ci-openshift-origin-master-e2e-gcp/1532383286834237440) that ran with these changes.  Look for the `: [sig-node] should not encounter ErrImagePull in non-openshift namespace pods` failure and notice the ErrImagePull failures associated with the 3 regexes in the PR are no longer present.

Look for the same failure in the [another CI job](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-ci-4.11-upgrade-from-stable-4.10-e2e-gcp-ovn-upgrade/1532093641084899328) that did not run with these changes and you can see what's different.

ref: [TRT-266](https://issues.redhat.com//browse/TRT-266)

### CODE REVIEW

**Comment by qJkee** [comment]
shouldn't we ignore all the errors given by that images?
I mean instead of checking for `3.7 in gcr.io/authenticated-image-pulling/alpine` check only image name `gcr.io/authenticated-image-pulling/alpine` since error messages may change

**Comment by DennisPeriquet** [comment]
> shouldn't we ignore all the errors given by that images? I mean instead of checking for `3.7 in gcr.io/authenticated-image-pulling/alpine` check only image name `gcr.io/authenticated-image-pulling/alpine` since error messages may change

Thanks for the comment.

Which error message do you think might change? maybe the "3.7" part of the current message?

The reason I went with such a specific match pattern (I did the same for the `webserver:404` image) was so that I would target very specific image names/tags as configured by the test in order to avoid skipping ErrImagePull messages that are real failures.

If the message from docker changed, this test would expose it as a failure that should be ignored; in this case, someone would have to look at and update this code to account for the new message (as you're probably pointing out).

But if I took out the "3.7" and the test added a "3.8" tag (that was supposed to result in a pullable image) and ended up with a real ErrImagePull event, we'd miss it.

In short, I'm err'ing on the side of avoiding the possibility of obscuring real failures.

File: pkg/synthetictests/kubelet.go [hunk 1]
```diff
@@ -462,6 +474,21 @@ func buildTestsFailIfRegexMatch(testName string, matchRE, dontMatchRE *regexp.Re
...
+		for i := 0; i < len(expectedErrPatterns); i++ {
+			if expectedErrPatterns[i].MatchString(estr) {
+				found = true
+				break
+			}
+		}
+		if found {
+			// ErrImagePull was found but also one of the expected error patters was found so don't
+			// count this as an ErrImagePull failure.
+			continue
```
Comment (stbenjam) [inline pkg/synthetictests/kubelet.go:489]: This is very readable so no need  to change anything but go supports labels with goto/continue/break:

```go
Outer:
for _, i := range outer {
  for _, j := range inner {
    if i == j {
      break Outer
    }
  }
}
```
